Sources/Tracing/NoOpTracer.swift

```diff
@@ -48,7 +48,7 @@ public struct NoOpTracer: LegacyTracerProtocol {
         // no-op
     }
 
-    public struct NoOpSpan: Span {
+    public struct NoOpSpan: Tracing.Span {
```

Comment:
This is one annoying spot, people MUST write Tracing.Span here

Reply:
What about swapping `TracerSpan` and `Span`?

Reply:
Maybe, what do @tomerd @fabianfett @Lukasa think?

Reply:
I don't find this to annoying. The tough part here is really only on the tracer implementers. TracerSpan is on all the users. If we want to keep the pain small, the first group should deal with it, not the second.

Reply:
Yeah it's on implementors to do the dance, I don't mind either I guess.

Reply:
I would prefer `TracerSpan` to `[Tracing.]Span` here. Swift’s name lookup is still tricky now (see https://forums.swift.org/t/pitch-fully-qualified-name-syntax/28482 for the context). If the client defines a top-level `Tracing` type, they will completely lose access to `Tracing.Span`. Not to say `TracerSpan` is shorter by two ASCII characters:)

Reply:
A typical tracing user doesn’t use either `[Tracing.]Span` or `Tracer.Span`, as the `Tracer.Span` type can be inferred from `Tracer.startSpan`.

Both `Span` and `TracerSpan` can cause some confusion with `Tracer.Span`, but the fact that `Span` is ambiguous, despite being possible to work around and comfort the compiler, will inevitably troubles code review.

Given that the type name is not used very often, I still suggests the longer but less error-prone `TracerSpan`.